vignettes/dev04_RECIPES.Rmd

```diff
@@ -0,0 +1,61 @@
+
+### The `blocks` object
+
+The `blocks` object is an array of `block` objects. The setup is very similar to blocks used in the Table Generator. The `blocks` object and the blocks from the Table Generator are two sides of the same coin, hence the naming convention used. Each `block` is a named list that can contain a number of different elements. Some of the most important elements are included in the example: `data`, `variable`, and `statistic`.
+
+The `data` element denotes which data set provided is being used to create the block. The `variable` element denotes the field or parameter from the data set used for the block's statistical computation. The `statistic` element denotes which statistic is being calculated for the field/parameter.
+
+A few other elements can also be present within a `block` object. The `recipes.json` file contains a couple more: `stat_arg` and `stat_options_fn`.  Currently no other elements have usage in the application, but active development will be changing this soon.
+
+## The parsing of `recipes.json`
+
+### The table object
+
+As stated previously, the key/name of the table object in `recipes.json` is assigned as a class when pulling the table object into R. This class is used to determine how the pre-filtering should be completed in the functions `filter_adsl()` and `filter_adae()`. New outputs can be created for a hypothetical table with key `tbl_01` by defining `filter_adsl.tbl_01()` and `filter_adae.tbl_01()`.
+
+The key/name of the table object will also be used to define the option in the recipe select input in the application.
+
+### The `blocks` object
+
+There is one additional sub-object not yet mentioned, the `recipe_inclusion` object, which can assign a class to the `blocks` object. The standard inclusion criteria is for all data sets in a `blocks` object to be present in the list of data sets uploaded and for all variables listed in the `block` objects to be present in the data set identified with them. Assigning a value to `recipe_inclusion` can affect the rules for inclusion in the recipes select input HTML element. If a user had a special set of criteria, they could assign `crit_01` to `recipe_inclusion` and assign the function `recipe_inclusion.crit_01()` to determine the criteria.
+
+### The `block` object
+
+Currently only one potential class is being assigned to the `block` object and it is the value of element `stat_options_fn`. For certain statistics, such as `MEAN`, a drop down is provided in the statistic block (e.g. `AVISIT` values for `MEAN`). In the event the user wants a block to output multiple weeks, they can provide a value for `stat_options_fn` and its corresponding function (i.e. A value of `opt_01` would correspond to `stat_options.opt_01()`). The user could also opt to hard code these values using the element `stat_options`.
```

Comment:
```suggestion
Currently, only one potential class is being assigned to the `block` object and it is the value of element `stat_options_fn`. If you recall, when the `MEAN` statistic is selected in the app, a dropdown list is displayed containing different `AVISIT` values. Thus, if the user wants a block to output multiple weeks, they can provide those values to `stat_options_fn`. Its corresponding function ( for example, with value of `Baseline` would correspond to `stat_options.Baseline()`). The user could also opt to hard code these values using the element `stat_options`.
```

Reply:
I tried to provide a more concrete example instead of using the `_01` method. How did I do?